File: Abstract Factory/Ruby/TeamAssessmentConnectionFactory.rb
require_relative "IConnectionFactory.rb"
require_relative "TeamAssessmentHTTPConnection.rb"
require_relative "TeamAssessmentRPCConnection.rb"

class TeamAssessmentConnectionFactory < IConnectionFactory
    def createHTTPConnection
        TeamAssessmentHTTPConnection.new
    end

    def createRPCConnection
        TeamAssessmentRPCConnection.new
    end
end

File: Abstract Factory/Javascript/TeamAssessmentConnectionFactory.js
import IConnectionFactory from "./IConnectionFactory.js";
import TeamAssessmentRPCConnection from "./TeamAssessmentRPCConnection.js";
import TeamAssessmentHTTPConnection from "./TeamAssessmentHTTPConnection.js";

class TeamAssessmentConnectionFactory extends IConnectionFactory {
  createRPCConnection() {
    return new TeamAssessmentRPCConnection();
  }

  createHTTPConnection() {
    return new TeamAssessmentHTTPConnection();
  }
}

export default TeamAssessmentConnectionFactory;